sympy/sets/sets.py

```diff
@@ -1202,28 +1202,32 @@ def reduce(args):
 
         # ===== Pair-wise Rules =====
         # Here we depend on rules built into the constituent sets
-        args = set(args)
+        args = FiniteSet(*args)
```

Comment:
Ping @asmeurer , @jksuom , @smichr . Any alternate way of doing [this](https://github.com/sympy/sympy/pull/11188/files#diff-7b3e9b0852f1b1d094f87233f56e201fR1205) ?